cmd/syncer/main.go

```diff
@@ -212,6 +215,46 @@ func initSyncerComponents(ctx context.Context, clusterFlavor cnstypes.CnsCluster
 				log.Errorf("failed to initialize nodeManager. Error: %+v", err)
 				os.Exit(1)
 			}
+			if configInfo.Cfg.Global.ClusterDistribution == "" {
+				config, err := rest.InClusterConfig()
+				if err != nil {
+					log.Errorf("failed to get InClusterConfig: %v", err)
+				}
+				clientset, err := kubernetes.NewForConfig(config)
+				if err != nil {
+					log.Errorf("failed to create kubernetes client with err: %v", err)
+				}
+
+				// Get the version info for the Kubernetes API server
+				versionInfo, err := clientset.Discovery().ServerVersion()
+				if err != nil {
+					log.Errorf("failed to fetch versionInfo with err: %v", err)
+				}
+
+				// Extract the version string from the version info
+				version := versionInfo.GitVersion
+				var ClusterDistNameToServerVersion = map[string]string{
+					"gke":       "Anthos",
+					"racher":    "Rancher",
+					"rke":       "Rancher",
+					"docker":    "DockerEE",
+					"dockeree":  "DockerEE",
+					"openshift": "Openshift",
+					"wcp":       "Supervisor",
+					"vmware":    "TanzuKubernetesCluster",
+					"unknown":   "VanillaK8S",
```

Comment:
why unknown is vanilla here?

Reply:
For native k8s installation the Server Version is empty. So, I'm using the key as unknown. However, it's open for discussion. Any suggestion on the key name ? How about "nativek8s" ?